### PR TITLE
protect for constraint fit

### PR DESCRIPTION
**Pull request by VinInn**
see https://its.cern.ch/jira/browse/CMSTRACK-125

added also a protection to not consider outlier the constraint itself: 
this is regression as it modifies the behavior even w/r/t 5_3

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
@nclopezo @degano 
David, Alessandro: what's happening with the bot?

**Comment by nclopezo** [comment]
@slava77 
Hi Slava, 
It was stuck, I woke it up

File: TrackingTools/TrackFitters/src/KFFittingSmoother.cc [hunk 1]
```diff
@@ -213,6 +215,9 @@ Trajectory KFFittingSmoother::fitOne(const TrajectorySeed& aSeed,
 	      LogTrace("TrackFitters") << "Rejecting outlier hit  with estimate " << cut << " at position "
 				       << j << " with rawId=" << myHits[j]->geographicalId().rawId();
 	      LogTrace("TrackFitters") << "The fit will be repeated without the outlier";
+              if (outlierDet==nullptr) {
```
Comment (VinInn) [inline TrackingTools/TrackFitters/src/KFFittingSmoother.cc]: opps

**Comment by slava77** [comment]
@cmsbuild please test

Vincenzo, is this the end of the fix?
I just wanted to check in case I missed something on the tracking hn.

**Comment by VinInn** [comment]
@slava77,
I hope this is it.
The two fix are independent though.
I will forward port as soon as this is fully approved

**Comment by davidlange6** [comment]
@slava77, @vuosalo - we are waiting for this PR to be signed for 734patch1. Can you confirm its needed?

**Comment by VinInn** [comment]
@mmusich @tlampen please confirm

**Comment by mmusich** [comment]
@VinInn @slava77 @vuosalo yes it is. 
@frmeier @diguida this is something you want to monitor as well

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/8342	5714979
based on visual code inspection (affects only special fitting situations) and no changes in jenkins test comparisons.
The actual issue fix is relying on Vincenzo's tests and Marco's confirmation.

**Comment by davidlange6** [comment]
+1

**Pull request merged**